docs/api.rst

```diff
@@ -6,7 +6,49 @@
 
 
 LIBBPF API
-==================
+==========
+
+Error Handling
+--------------
+
+When libbpf is used in "libbpf 1.0 mode",API functions can return errors in one of two ways.
+
+You can set "libbpf 1.0" mode with the following line:
+
+```
+libbpf_set_strict_mode(LIBBPF_STRICT_DIRECT_ERRS | LIBBPF_STRICT_CLEAN_PTRS);
+```
+
+If the function returns an error code directly, it uses 0 to indicate success
+and a negative error code to indicate what caused the error. In this case the
+error code should be checked directly from the return, you do not need to check
+errno. 
+
+For example:
+
+```
+    err = some_libbpf_api_with_error_return(...);
+    if (err < 0) {
+        /* Handle error accordingly */
```

Comment:
hm... I think this example doesn't show that both err and errno can be used, let's add this snippet:

```
...
if (err < 0) {
    if (err = -EINVAL) {
        /* handle EINVAL error */
    } else if (errno == E2BIG) { /* note no minus sign */
        /* handle E2BIG error */
    }
```

This should make clear that both err and errno have error code set (and hopefully people will notice that errno has positive error code, while err is negative)